WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingsViewController.swift

```diff
@@ -0,0 +1,374 @@
+            return blogReuseIdentifier
+        default:
+            return defaultReuseIdentifier
+        }
+    }
+    
+    private func configureCell(cell: UITableViewCell, indexPath: NSIndexPath) {
+        let settings = settingsForRowAtIndexPath(indexPath)!
+        
+        switch settings.channel {
+        case let .Blog(blogId) where isLoadMoreRow(indexPath):
+            cell.textLabel?.text            = NSLocalizedString("View all...", comment: "Displays More Rows")
+            cell.textLabel?.textAlignment   = .Center
+            cell.accessoryType              = .None
+            WPStyleGuide.configureTableViewCell(cell)
+        case let .Blog(blogId):
+            cell.textLabel?.text            = settings.blog?.blogName ?? settings.channel.description()
+            cell.detailTextLabel?.text      = settings.blog?.displayURL ?? String()
+            cell.accessoryType              = .DisclosureIndicator
+            cell.imageView?.setImageWithSiteIcon(settings.blog?.icon)
+            WPStyleGuide.configureTableViewSmallSubtitleCell(cell)
+        default:
+            cell.textLabel?.text            = settings.channel.description()
+            cell.textLabel?.textAlignment   = .Left
+            cell.accessoryType              = .DisclosureIndicator
+            WPStyleGuide.configureTableViewCell(cell)
+        }
+    }
+    
+    private func settingsForRowAtIndexPath(indexPath: NSIndexPath) -> NotificationSettings? {
+        return groupedSettings?[indexPath.section][indexPath.row]
+    }
+    
+    private func isSectionEmpty(sectionIndex: Int) -> Bool {
+        return groupedSettings == nil || groupedSettings?[sectionIndex].count == 0
+    }
+    
+    
+    
+    // MARK: - Load More Helpers
+    private func displaysLoadMoreRow() -> Bool {
+        return groupedSettings?[Section.Blog.rawValue].count > loadMoreRowIndex && !displayMoreWasAccepted
+    }
+    
+    private func isLoadMoreRow(indexPath: NSIndexPath) -> Bool {
+        if displaysLoadMoreRow() {
+            return indexPath.section == Section.Blog.rawValue && indexPath.row == loadMoreRowIndex
+        }
+        
+        return false
+    }
+    
+    private func displayMoreBlogs() {
+        // Remember this action!
+        displayMoreWasAccepted = true
+        
+        // And refresh the section
+        let sections = NSIndexSet(index: Section.Blog.rawValue)
+        tableView.reloadSections(sections, withRowAnimation: .Fade)
+    }
+
+    
+    
+    // MARK: - Segue Helpers
+    private func displayDetailsForSettings(settings: NotificationSettings) {
+        switch settings.channel {
+        case .WordPressCom:
+            // WordPress.com Row will push the SettingDetails ViewController, directly
+            let detailsViewController = NotificationSettingDetailsViewController(style: .Grouped)
+            detailsViewController.setupWithSettings(settings, stream: settings.streams.first!)
```

Comment:
Thought of doing this. Problem with it is that if you wanna move over to a Storyboard (which is something we'll likely end up doing), you just cannot call the designated initializer.

In that case, UIKit will always call the `NSCoder` initializer. Hoooowever, let's add a _convenience_ initializer, and keep the `setupWith...` method. That way we keep our options open, and the initialization syntax remains compact.

Thank you!